src/lightning/pytorch/loops/training_epoch_loop.py

```diff
@@ -165,6 +165,7 @@ def reset(self) -> None:
 
     def on_run_start(self, data_fetcher: _DataFetcher) -> None:
         iter(data_fetcher)  # creates the iterator inside the fetcher
```

Comment:
Should this check if we are on the first iteration, since then the `fit_loop` will have already `iter`ed it?

Reply:
I guarded this with `trainer.current_epoch > 0` and updated the "worker-shutdown test"

Reply:
That won't be right when restarting from a checkpoint. What if you check if the inner iterator is setup already? Or you could use `loop.restarting` instead.